src/Core.Nhibernate/Extensions/IdentityServerServiceFactoryExtensions.cs

```diff
@@ -1,91 +1,65 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-using FluentNHibernate.Automapping;
-using FluentNHibernate.Cfg;
-using FluentNHibernate.Cfg.Db;
-using FluentNHibernate.Conventions.Helpers;
-using IdentityServer3.Core.Configuration;
 using IdentityServer3.Core.Services;
 using NHibernate;
-using NHibernate.Cfg;
-using NHibernate.Tool.hbm2ddl;
-using System.Configuration;
-using Core.Nhibernate;
-using IdentityServer3.Contrib.Nhibernate;
 using IdentityServer3.Contrib.Nhibernate.Services;
 using IdentityServer3.Contrib.Nhibernate.Stores;
+using IdentityServer3.Core.Models;
 
 namespace IdentityServer3.Core.Configuration
 {
     public static class IdentityServerServiceFactoryExtensions
     {
 
         public static void RegisterNhibernateStores(this IdentityServerServiceFactory factory,
-            NhibernateServiceOptions serviceOptions)
+            ISessionFactory nHibernateSessionFactory,
+            IDbProfileConfig dbProfileConfig, 
+            bool registerOperationalServices = false, 
+            bool registerConfigurationServices = false)
```

Comment:
I would prefer to keep the parameters object. It makes the API easier to use because it's harder to confuse the 2 bool parameters.
The AutoMapper piece could be added to it.